Do not print stack trace if 'which' cannot be found

### Description of the bug
If you run a project on a Linux without `which` you see the following output
```
[INFO] --- vaadin-maven-plugin:18.0.0.alpha1:prepare-frontend (default) @ starter-wizard ---
[ERROR] Failed to execute the command '[which, node]'
java.io.IOException: Cannot run program "which": error=2, No such file or directory
    at java.lang.ProcessBuilder.start (ProcessBuilder.java:1142)
    at java.lang.ProcessBuilder.start (ProcessBuilder.java:1073)
    at com.vaadin.flow.server.frontend.FrontendToolsLocator.executeCommand (FrontendToolsLocator.java:120)
    at com.vaadin.flow.server.frontend.FrontendToolsLocator.tryLocateTool (FrontendToolsLocator.java:75)
    at com.vaadin.flow.server.frontend.FrontendTools.lambda$getExecutable$2 (FrontendTools.java:507)
    at java.util.Optional.orElseGet (Optional.java:362)
    at com.vaadin.flow.server.frontend.FrontendTools.getExecutable (FrontendTools.java:507)
    at com.vaadin.flow.server.frontend.FrontendTools.getNodeExecutable (FrontendTools.java:228)
    at com.vaadin.flow.server.frontend.FrontendTools.validateNodeAndNpmVersion (FrontendTools.java:312)
    at com.vaadin.flow.plugin.maven.PrepareFrontendMojo.execute (PrepareFrontendMojo.java:111)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:64)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.io.IOException: error=2, No such file or directory
    at java.lang.ProcessImpl.forkAndExec (Native Method)
    at java.lang.ProcessImpl.<init> (ProcessImpl.java:313)
    at java.lang.ProcessImpl.start (ProcessImpl.java:244)
    at java.lang.ProcessBuilder.start (ProcessBuilder.java:1109)
    at java.lang.ProcessBuilder.start (ProcessBuilder.java:1073)
    at com.vaadin.flow.server.frontend.FrontendToolsLocator.executeCommand (FrontendToolsLocator.java:120)
    at com.vaadin.flow.server.frontend.FrontendToolsLocator.tryLocateTool (FrontendToolsLocator.java:75)
    at com.vaadin.flow.server.frontend.FrontendTools.lambda$getExecutable$2 (FrontendTools.java:507)
    at java.util.Optional.orElseGet (Optional.java:362)
    at com.vaadin.flow.server.frontend.FrontendTools.getExecutable (FrontendTools.java:507)
    at com.vaadin.flow.server.frontend.FrontendTools.getNodeExecutable (FrontendTools.java:228)
    at com.vaadin.flow.server.frontend.FrontendTools.validateNodeAndNpmVersion (FrontendTools.java:312)
    at com.vaadin.flow.plugin.maven.PrepareFrontendMojo.execute (PrepareFrontendMojo.java:111)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:64)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[INFO] Couldn't find node. Installing Node and NPM to /home/myuser/.vaadin.
[INFO] Installing node version v12.18.3
```

### Expected behavior
The following would be enough

```
[INFO] --- vaadin-maven-plugin:18.0.0.alpha1:prepare-frontend (default) @ starter-wizard ---
[ERROR] Failed to execute the command '[which, node]'
[INFO] Couldn't find node. Installing Node and NPM to /home/myuser/.vaadin.
[INFO] Installing node version v12.18.3
```

--- a/flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendToolsLocator.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendToolsLocator.java
@@ -119,7 +119,13 @@ public class FrontendToolsLocator implements Serializable {
             process = FrontendUtils
                     .createProcessBuilder(Arrays.asList(commandParts)).start();
         } catch (IOException e) {
-            log().error("Failed to execute the command '{}'", commandString, e);
+            if (logErrorOnFail) {
+                log().error("Failed to execute the command '{}'", commandString,
+                        e);
+            } else if (log().isDebugEnabled()) {
+                log().debug("Failed to execute the command '{}'", commandString,
+                        e);
+            }
             return Optional.empty();
         }
 

